tests/async/tjsandnativeasync.nim

```diff
@@ -0,0 +1,30 @@
+discard """
+  output: '''hi
+bye'''
+"""
+
+import async, times
+when defined(js):
+    proc sleepAsync(t: int): Future[void] =
```

Comment:
This should be in ``asyncjs``, no?

Reply:
I don't think so. `sleepAsync`, or better `setTimeout` should be a function of `dispatcher` whichever is "current", so that one can implement custom runloops. While `async` macro and `Future[T]` are completely platform independent concepts.